docs/csharp/language-reference/keywords/is.md

```diff
@@ -31,35 +32,135 @@ translation.priority.ht:
   - "zh-cn"
   - "zh-tw"
 ---
-# is (C# Reference)
-Checks if an object is compatible with a given type. For example, the following code can determine if an object is an instance of the `MyObject` type, or a type that derives from `MyObject`:  
+# is (C# Reference) #
+
+Checks if an object is compatible with a given type, or (starting with C# 7) tests an expression against a pattern.
+
+## Testing for type compatibility ##
+
+The `is` keyword evaluates type compatibility at runtime. It determines whether an object instance or the result of an expression can be converted to a specified type. It has the syntax
+
+```cs
+   expr is type
+```
+
+where *expr* is an expression that evaluates to an instance of some type, and *type* is the name of the type to which the result of *expr* is to be converted. The `is` statement is `true` if *expr* is non-null and the object that results from evaluating the expression can be converted to *type*; otherwise, it returns `false`.
+
+For example, the following code determines if `obj` can be cast to an instance of the `Person` type:
+
+[!code-cs[is#1](../../../../samples/snippets/csharp/language-reference/keywords/is/is1.cs)]
+
+The `is` statement is true if:
+
+- *expr* is an instance of the same type as *type*.
+
+- *expr* is an instance of a type that derives from *type*. In other words, the result of *expr* can be upcast to an instance of *type*.
+
+- *expr* is an instance of a type that is a base class of *type*, and *expr* is the result of a downcast from *type* or one of its derived types.
```

Comment:
@BillWagner I've revised it. Incredibly, I couldn't find a definition of compile-time type and runtime type anywhere in our doc set (other than in a blog). I think I'll open an issue so that, at some point, its absence can be addressed in the docs.